src/cryptonote_core/blockchain.cpp

```diff
@@ -1217,7 +1244,7 @@ bool Blockchain::create_block_template(block& b, const account_public_address& m
    */
   //make blocks coin-base tx looks close to real coinbase tx to get truthful blob size
   uint8_t hf_version = m_hardfork->get_current_version();
-  size_t max_outs = hf_version >= 4 ? 1 : 11;
+  size_t max_outs = hf_version >= HF_VERSION_ALLOW_RCT ? 1 : 11;
```

Comment:
The hf versions of allowing/enforcing RCT have been merged intentionally, yes.

Reply:
Is the original intent here to set `max_outs` to 1 when RCT is enforced? If so, the number should've been 6 instead of 4, no?

Currently HF_VERSION_ALLOW_RCT and HF_VERSION_ENFORCE_RCT are both set to 255, meaning the hf version for activating RCT is undetermined currently, but this doesn't necessarily mean the hf versions of RCT adoption and RCT enforcement should be the same.

If my above observation is correct, we should use `HF_VERSION_ENFORCE_RCT` here, shouldn't we?

Reply:
Now I think it's fine to leave it as is.